web/src/components/lineage/components/edge/Edge.tsx

```diff
@@ -1,19 +1,39 @@
 // Copyright 2018-2023 contributors to the Marquez project
 // SPDX-License-Identifier: Apache-2.0
 
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { GraphEdge } from 'dagre'
 import { LinePath } from '@visx/shape'
 import { curveMonotoneX } from '@visx/curve'
+import { faCaretRight } from '@fortawesome/free-solid-svg-icons/faCaretRight'
 import { theme } from '../../../../helpers/theme'
 import React from 'react'
 
 type EdgeProps = {
   edgePoints: GraphEdge[]
 }
 
+const RADIUS = 14
+const OUTER_RADIUS = RADIUS + 8
+const ICON_SIZE = 16
+
 class Edge extends React.Component<EdgeProps> {
+  getPoints = (edge: any) => edge.points[edge.points.length - 1]
+
   render() {
     const { edgePoints } = this.props
+    const edgeEnds = edgePoints.map(edge => {
+      const isSelected = edgePoints.find(
+        o =>
+          this.getPoints(o).x == this.getPoints(edge).x &&
+          this.getPoints(o).y == this.getPoints(edge).y &&
+          o.isSelected === true
+      )
+      return Object.assign(edge.points[edge.points.length - 1], {
+        isSelected: typeof isSelected !== 'undefined'
+      })
```

Comment:
I think you can use the spread operator for things like this to make it more readable.

Reply:
No problem, did it